ffi/src/transaction/mod.rs

```diff
@@ -94,6 +94,23 @@ pub unsafe extern "C" fn add_files(
     txn.add_files(write_metadata);
 }
 
+///
+/// Mark the transaction as having data changes or not (these are recorded at the file level).
+///
+/// When set here writers should not provide a data_change column in EngineData.
+///
+/// # Safety
+///
+/// Caller is responsible for passing a valid handle. CONSUMES TRANSACTION and commit info
```

Comment:
These apis that consume the argument are inherently more dangerous because there's no compiler on the other side enforcing things like there in pure rust.

Could we just `as_ref_mut()` it instead, and set the bool on the txn? I think that might require a new method in `handle.rs` but it should be pretty trivial to write that (and if it's not stop and we can do it this way)

Reply:
I added a a new set_data_change method on Transaction and called it here (renaming this method as well) to avoid consuming the value.

Please let me know if this is what you had in mind.  It seems at least at the surface `as_mut` on handle works here but I guess there might be subtleties